lms/djangoapps/certificates/signals.py

```diff
@@ -1,29 +1,25 @@
-""" Signal handler for enabling self-generated certificates by default
-for self-paced courses.
+""" Signal handler for enabling self-generated certificates for self-paced
+courses and disabling for instructor-paced courses.
 """
 from celery.task import task
 from django.dispatch.dispatcher import receiver
 
 from certificates.models import CertificateGenerationCourseSetting
 from opaque_keys.edx.keys import CourseKey
-from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
-from xmodule.modulestore.django import SignalHandler
+from xmodule.modulestore.django import SignalHandler, modulestore
 
 
 @receiver(SignalHandler.course_published)
 def _listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable=unused-argument
     """ Catches the signal that a course has been published in Studio and
-    enable the self-generated certificates by default for self-paced
-    courses.
+    enable/disable the self-generated certificates for courses according to pacing.
     """
-    enable_self_generated_certs.delay(unicode(course_key))
+    set_self_generated_certs.delay(unicode(course_key))
 
 
 @task()
-def enable_self_generated_certs(course_key):
-    """Enable the self-generated certificates by default for self-paced courses."""
+def set_self_generated_certs(course_key):
```

Comment:
But enable_self_generated_certs only shows that we are enabling that's why i changed method name.